python/ql/src/experimental/Security/CWE-074/paramiko/paramiko.ql

```diff
@@ -0,0 +1,72 @@
+/**
+ * @name RCE with user provided command with paramiko ssh client
+ * @description user provided command can lead to execute code on a external server that can be belong to other users or admins
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity 9.3
+ * @precision high
+ * @id py/command-injection
+ * @tags security
+ *       experimental
+ *       external/cwe/cwe-074
+ */
+
+import python
+import semmle.python.dataflow.new.DataFlow
+import semmle.python.dataflow.new.TaintTracking
+import semmle.python.dataflow.new.RemoteFlowSources
+import semmle.python.ApiGraphs
+import DataFlow::PathGraph
+
+class ParamikoCMDInjectionConfiguration extends TaintTracking::Configuration {
+  ParamikoCMDInjectionConfiguration() { this = "ParamikoCMDInjectionConfiguration" }
+
+  override predicate isSource(DataFlow::Node source) { source instanceof RemoteFlowSource }
+
+  override predicate isSink(DataFlow::Node sink) {
+    sink =
+      [
+        API::moduleImport("paramiko")
+            .getMember("SSHClient")
+            .getReturn()
+            .getMember("exec_command")
+            .getACall()
+            .getArgByName("command"),
+        API::moduleImport("paramiko")
+            .getMember("SSHClient")
+            .getReturn()
+            .getMember("exec_command")
+            .getACall()
+            .getArg(0)
+      ]
+    or
+    sink =
+      [
+        API::moduleImport("paramiko")
+            .getMember("SSHClient")
+            .getReturn()
+            .getMember("connect")
+            .getACall()
+            .getArgByName("sock"),
+        API::moduleImport("paramiko")
+            .getMember("SSHClient")
+            .getReturn()
+            .getMember("connect")
+            .getACall()
+            .getArg(11)
+      ]
+  }
```

Comment:
This code is fine as is, but I would refactor it through a private predicate to make it clearer what is happening.
Also, we can use the member predicate `getParameter` that takes both an index and a name:
```suggestion
  private API::Node paramikoClient() {
    result = API::moduleImport("paramiko").getMember("SSHClient").getReturn()
  }
  
  override predicate isSink(DataFlow::Node sink) {
    sink = paramikoClient().getMember("exec_command").getACall().getParameter(0, "command").asSink()
    or
    sink = paramikoClient().getMember("connect").getACall().getParameter(11, "sock").asSink()
  }
```
(you may need to run autoformat to break the lines.)